routes/candidateRoutes.js

```diff
@@ -121,13 +137,30 @@ candidateRouter.put('/update/id=:id', authorizedAdmin, async (req, res) => {
 
 candidateRouter.delete('/delete/id=:id', authorizedAdmin, async (req, res) => {
     const candidateId = req.params.id;
-    Candidate.deleteOne({ id: Number(candidateId) }).then(() => {
+    //Getting the image url from db before deleting it
+    const findImgUrl = await Candidate.findOne({ id: Number(candidateId) });
+    if(findImgUrl){
+        const imgRegex = /[^/]+$/;
+        const imgUrl = RegExp(imgRegex).exec(findImgUrl.img);
+        const params = {
+            Bucket: BUCKET_NAME,
+            Key: imgUrl[0],
+        };
+        const command = new DeleteObjectCommand(params);
+        const imgDeleted = await s3.send(command);
+        if(!imgDeleted){
+            logger.error("Unable to delete image from s3");
+            return res.status(400).send({error: 'Unable to delete image'});
```

Comment:
How is this a bad request? This is totally Amazon failing to delete the image, user did not make a bad request

Reply:
Internal server error then?